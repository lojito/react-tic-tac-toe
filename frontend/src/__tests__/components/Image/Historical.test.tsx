import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render, screen } from "@testing-library/react";
import ImageHistorical from "../../../components/Image/Historical";
import {
  DictionaryContext,
  DictionaryEntry,
} from "../../../contexts/dictionary/DictionaryContext";

describe("ImageHistorical", () => {
  const LIONEL_MESSI = 13;
  const CRISTIANO_RONALDO = 15;

  const renderImageHistorical = () => {
    const mockDictionary = {
      IMAGE_USER_LABEL: "You",
      IMAGE_COMPUTER_LABEL: "PC",
    } as DictionaryEntry;

    const utils = render(
      <DictionaryContext.Provider value={{ dictionary: mockDictionary }}>
        <ImageHistorical
          folder="soccer"
          userImage={LIONEL_MESSI}
          computerImage={CRISTIANO_RONALDO}
        />
      </DictionaryContext.Provider>
    );

    const divWrapper = screen.getByTestId("images-historical");
    const userImage = screen.getByRole("img", { name: "user" });
    const computerImage = screen.getByRole("img", { name: "computer" });
    const labelUserImage = screen.getByText(
      `${mockDictionary.IMAGE_USER_LABEL}:`
    );
    const labelComputerImage = screen.getByText(
      `${mockDictionary.IMAGE_COMPUTER_LABEL}:`
    );

    return {
      ...utils,
      mockDictionary,
      divWrapper,
      userImage,
      computerImage,
      labelUserImage,
      labelComputerImage,
    };
  };

  it("should render correctly", () => {
    const { asFragment } = renderImageHistorical();

    expect(asFragment()).toMatchSnapshot();
  });

  it("should apply the 'images' and 'historical' classes", () => {
    const { divWrapper } = renderImageHistorical();

    expect(divWrapper).toHaveClass("images");
    expect(divWrapper).toHaveClass("historical");
  });

  it("should find the text 'You:' in the document", () => {
    const { labelUserImage } = renderImageHistorical();

    expect(labelUserImage).toBeInTheDocument();
  });

  it("should find the text 'PC:' in the document", () => {
    const { labelComputerImage } = renderImageHistorical();

    expect(labelComputerImage).toBeInTheDocument();
  });

  it("should find the user image in the document", () => {
    const { userImage } = renderImageHistorical();

    expect(userImage).toBeInTheDocument();
  });

  it("should find the computer image in the document", () => {
    const { computerImage } = renderImageHistorical();

    expect(computerImage).toBeInTheDocument();
  });
});
