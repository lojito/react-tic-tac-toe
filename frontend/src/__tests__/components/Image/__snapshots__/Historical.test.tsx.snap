// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageHistorical should render correctly 1`] = `
<DocumentFragment>
  <div
    class="images historical"
    data-testid="images-historical"
  >
    <div
      class="user-image"
    >
      <span>
        You: 
      </span>
      <img
        alt="user"
        class="image-token"
        src="undefined/images/soccer/13.jpg"
      />
    </div>
    <div
      class="computer-image"
    >
      <span>
        PC: 
      </span>
      <img
        alt="computer"
        class="image-token"
        src="undefined/images/soccer/15.jpg"
      />
    </div>
  </div>
</DocumentFragment>
`;
